### PR TITLE
KTOR-3799 Add charset only for text/* content types

### PR DESCRIPTION
**Pull request by rsinukov**
https://youtrack.jetbrains.com/issue/KTOR-3799

### CODE REVIEW

**Comment by e5l** [comment]
Please check compilation errors on linux

File: ktor-http/common/src/io/ktor/http/ContentTypes.kt [hunk 1]
```diff
@@ -286,7 +286,7 @@ public fun ContentType.withCharset(charset: Charset): ContentType =
  * if [ContentType] is not ignored
  */
 public fun ContentType.withCharsetIfNeeded(charset: Charset): ContentType =
-    if (contentType.lowercase() == "application" && contentSubtype.lowercase() == "json") {
+    if (contentType.lowercase() != "text") {
```
Comment (Stexxe) [inline ktor-http/common/src/io/ktor/http/ContentTypes.kt:289]: This is a workaround replaced with a more general workaround.

Comment (rsinukov) [reply ktor-http/common/src/io/ktor/http/ContentTypes.kt:289]: No, it's not. Only text content type should have charset parameter.

**Comment by rsinukov** [comment]
@e5l thanks, fixed. It was an imports problem

**Review by e5l — APPROVED** [review]

File: ktor-http/common/src/io/ktor/http/ContentTypes.kt (hunk 1 above)
Comment (Stexxe) [reply ktor-http/common/src/io/ktor/http/ContentTypes.kt:289]: You right. Although, I don't like the method's name. `withCharsetIfApplicable` would be better.

**Pull request merged**